src/ui-layout.js

```diff
@@ -429,36 +474,48 @@ angular.module('ui.layout', [])
       var prevContainer = ctrl.containers[index-2];
       var isLastContainer = index === (ctrl.containers.length - 1);
       var endDiff;
+      var flowProperty = ctrl.sizeProperties.flowProperty;
+      var sizeProperty = ctrl.sizeProperties.sizeProperty;
 
       ctrl.bounds = $element[0].getBoundingClientRect();
 
       c.collapsed = !ctrl.containers[index].collapsed;
 
+      // uncollapsedSize is undefined in case of 'auto' sized containers.
+      // Perhaps there's a place where we could set... could find it though. @see also toggleBefore
+      if (c.uncollapsedSize === undefined) {
+        c.uncollapsedSize = c.size;
+      } else {
+        c.uncollapsedSize = parseInt(c.uncollapsedSize);
+      }
+
+      // FIXME: collapse:resize:uncollapse: works well "visually" without the prevSplitbar and prevContainer calculations
+      // but removing those breaks few test
       $scope.$apply(function() {
         if(c.collapsed) {
-          c.actualSize = c.size;
+
           c.size = 0;
 
           // adds additional space so the splitbar moves to the very end of the container
           // to offset the lost space when converting from percents to pixels
-          endDiff = (isLastContainer) ? ctrl.bounds[ctrl.sizeProperties.sizeProperty] - c[ctrl.sizeProperties.flowProperty] - c.actualSize : 0;
+          endDiff = (isLastContainer) ? ctrl.bounds[sizeProperty] - c[flowProperty] - c.uncollapsedSize : 0;
 
-          if(prevSplitbar) prevSplitbar[ctrl.sizeProperties.flowProperty] += (c.actualSize + endDiff);
-          if(prevContainer) prevContainer.size += (c.actualSize + endDiff);
+          if(prevSplitbar) prevSplitbar[flowProperty] += (c.uncollapsedSize + endDiff);
+          if(prevContainer) prevContainer.size += (c.uncollapsedSize + endDiff);
 
         } else {
-          c.size = c.actualSize;
+          c.size = c.uncollapsedSize;
 
           // adds additional space so the splitbar moves back to the proper position
           // to offset the additional space added when collapsing
-          endDiff = (isLastContainer) ? ctrl.bounds[ctrl.sizeProperties.sizeProperty] - c[ctrl.sizeProperties.flowProperty] - c.actualSize : 0;
+          endDiff = (isLastContainer) ? ctrl.bounds[sizeProperty] - c[flowProperty] - c.uncollapsedSize : 0;
 
-          if(prevSplitbar) prevSplitbar[ctrl.sizeProperties.flowProperty] -= (c.actualSize + endDiff);
-          if(prevContainer) prevContainer.size -= (c.actualSize + endDiff);
+          if(prevSplitbar) prevSplitbar[flowProperty] -= (c.uncollapsedSize + endDiff);
```

Comment:
This would be more readable with brackets {}.
Same thing goes for the line under neath it.